### PR TITLE
Reduce the number of dependencies

### PR DESCRIPTION
**Pull request by arkgil**
Hey folks 👋 It's really great to see Elixir being supported as one of the first languages for Test Analytics!

I stumbled into a bunch of dependency conflicts when installing the collector, and I found that that telemetry is not used at all, and Ecto is only used for UUID generation. I removed telemetry altogether and replaced Ecto with a dedicated UUID library. 

Let me know what you think!

### CODE REVIEW

File: lib/buildkite_test_collector/test_result.ex [hunk 1]
```diff
@@ -30,7 +29,7 @@ defmodule BuildkiteTestCollector.TestResult do
 
   @typedoc "Individual test summary.  Spec as yet unconfirmed."
   @type t :: %TestResult{
-          id: UUID.t(),
+          id: String.t(),
```
Comment (arkgil) [inline lib/buildkite_test_collector/test_result.ex:32]: Made it a string, but I can also create a custom type for the UUID if you prefer that.

**Comment by joshprice** [comment]
Thanks so much for the PR! 👍🏻 

We'll be adding telemetry for Ecto shortly, so although we don't need these now, we will very shortly.

We also explicitly chose Ecto's UUID because https://github.com/zyro/elixir-uuid is not currently maintained. It's last commit was 5 years ago and there are many unmerged PRs and unaddressed issues, like this one https://github.com/zyro/elixir-uuid/issues/60

What were the dependency conflicts you ran into?

**Comment by joshprice** [comment]
Just found a thread on the UUID package (https://github.com/zyro/elixir-uuid/issues/50) which mentions a potential UUID alternative we could investigate

https://github.com/bitwalker/uniq

**Comment by jimsynz** [comment]
@joshprice we could perhaps make it an optional dependency?

**Comment by arkgil** [comment]
Ah, that makes sense. Monitoring Ecto in tests sounds pretty interesting!

One thing you could consider is relaxing the requirement on Telemetry. There are no API changes between version 0.4 and 1.0, so having the ability to use either of the two would definitely help with adoption. See e.g. Phoenix: https://github.com/phoenixframework/phoenix/blob/0336846f27cb366c76c2b03f2bca220d25a56e08/mix.exs#L71, and Ecto: https://github.com/elixir-ecto/ecto/blob/1f04999ab2647115d75140d35840b30e40be60db/mix.exs#L35.

**Comment by arkgil** [comment]
Actually, relaxing the requirement on Ecto version would help too, since 3.8 is pretty recent.

**Comment by joshprice** [comment]
Closing this PR. As @jimsynz pointed out we'll look at making the deps optional and relaxing the versions as much as possible, as not everyone will be tracing SQL. Thanks again for the valuable feedback, most appreciated! 🙏🏻

**Pull request closed without merging**